### PR TITLE
Fix -Wattributes warning

### PR DESCRIPTION
**Pull request by jschueller**
- fix a warning
- use add_test with NAME/COMMAND signautre to fix cross compilation

### CODE REVIEW

File: include/pagmo/archipelago.hpp [hunk 1]
```diff
@@ -140,7 +140,7 @@ PAGMO_DLL_PUBLIC std::ostream &operator<<(std::ostream &, migrant_handling);
 class PAGMO_DLL_PUBLIC archipelago
 {
     // Make friends with island.
-    friend class PAGMO_DLL_PUBLIC island;
+    friend class island;
```
Comment (bluescarni) [inline include/pagmo/archipelago.hpp:143]: Could you explain what kind of warnings the ``PAGMO_DLL_PUBLIC`` attribute is giving?

As far as I remember, the original motivation for repeating ``PAGMO_DLL_PUBLIC`` here was that MSVC would complain if a class is not declared everywhere with the same exact set of attributes. But perhaps I am misremembering...

Comment (jschueller) [reply include/pagmo/archipelago.hpp:143]: its the Wattributes warning
I dont see any warnings in the windows compilation log

Comment (jschueller) [reply include/pagmo/archipelago.hpp:143]: maybe it was for an older version of msvc ?

Comment (bluescarni) [reply include/pagmo/archipelago.hpp:143]: Possibly... in any case, sorry for taking this long, merging this now.

**Pull request merged**